tests/model_registry/model_catalog/conftest.py

```diff
@@ -74,7 +76,20 @@ def enabled_model_catalog_config_map(
     patches = {"data": {"sources.yaml": enabled_sources_yaml}}
 
     with ResourceEditor(patches={user_sources_cm: patches}):
+        # Wait fot the model catalog pod to be ready
         is_model_catalog_ready(client=admin_client, model_registry_namespace=model_registry_namespace)
+
+        # Wait for the model catalog API to be fully ready
+        # Note: Build URL and headers directly to maintain session scope
+        if model_catalog_routes := list(
+            Route.get(
+                namespace=model_registry_namespace, label_selector="component=model-catalog", dyn_client=admin_client
+            )
+        ):
+            catalog_url = f"https://{model_catalog_routes[0].instance.spec.host}:443/api/model_catalog/v1alpha1/"
+            headers = get_rest_headers(token=current_client_token)
+            wait_for_model_catalog_api(url=catalog_url, headers=headers)
+
```

Comment:
You can make it a utility and use retry decorator on it. Because if it is not ready, we should not proceed. What do you think?